proto/pulumi/resource.proto

```diff
@@ -238,3 +239,11 @@ message TransformResponse {
     google.protobuf.Struct properties = 1; // the transformed input properties.
     TransformResourceOptions options = 2; // the options for the resource.
 }
+
+message RegisterDefaultProvidersRequest {
```

Comment:
I wonder if we should name this something more general? 
We could use the same trick for something like contextual parents as well.

Reply:
Yeah, the same trick could work in other places. Should the parameters then be a union type instead?  Or would it make sense to just have different grpc methods for it?  Maybe `CreateNewContext` would work for a unionized call?

Reply:
I think just a message where all the fields are optional (or zero) would work fine.